report_qweb_pdf_cover/models/ir_actions_report.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2024 ForgeFlow S.L. (https://www.forgeflow.com)
+# Part of ForgeFlow. See LICENSE file for full copyright and licensing details.
+
+from base64 import b64decode
+from io import BytesIO
+from logging import getLogger
+
+from PyPDF2 import PdfFileReader, PdfFileWriter
+
+from odoo import _, api, fields, models
+
+logger = getLogger(__name__)
+
+
+class Report(models.Model):
+    _inherit = "ir.actions.report"
+
+    use_front_cover = fields.Boolean(
+        default=False, help="Use a front cover when rendering the PDF report."
+    )
+    use_back_cover = fields.Boolean(
+        default=False, help="Use a back cover when rendering the PDF report."
+    )
+
+    front_cover_overlap = fields.Boolean(
+        default=False,
+        string="Overlap Front Cover",
+        help="When set, the front cover of the report will overlap with the "
+        "contents of the first page of the report. This is useful to include "
+        "some information of the report in the front cover.",
+    )
+    back_cover_overlap = fields.Boolean(
+        default=False,
+        string="Overlap Back Cover",
+        help="When set, the back cover of the report will overlap with the "
+        "contents of the last page of the report. This is useful to include "
+        "some information of the report in the back cover.",
+    )
+
+    front_cover_pdf = fields.Binary(
+        string="Front Cover PDF",
+        help="Upload an PDF file to use as a front cover on this report.",
+    )
+    back_cover_pdf = fields.Binary(
+        string="Back Cover PDF",
+        help="Upload an PDF file to use as a back cover on this report.",
+    )
+
+    @api.model
+    def pdf_check_pages(self, num_pages, front=True):
+        if num_pages < 1:
+            if front:
+                logger.error(_("Your front cover PDF does not contain any pages."))
```

Comment:
```suggestion
                logger.error(self.env._("Your front cover PDF does not contain any pages."))
```

This is better performance. Can you please replace it in all '_' occurrences?

Reply:
Updated, Thanks